Maple2.Model/Game/Event/AttendGift.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Maple2.Model.Enum;
+using Maple2.PacketLib.Tools;
+using Maple2.Tools;
+using Maple2.Tools.Extensions;
+
+namespace Maple2.Model.Game.Event;
+
+/// <summary>
+/// Attendance Event
+/// </summary>
+public class AttendGift : GameEventInfo {
+    public string AttendanceName { get; init; } = string.Empty;
+    public string Url { get; init; } = string.Empty;
+    public long BeginTime { get; init; }
+    public long EndTime { get; init; }
+    public bool DisableClaimButton { get; init; }
+    public int TimeRequired { get; init; }
+    public AttendGiftCurrencyType SkipDayCurrencyType { get; init; }
+    public int SkipDaysAllowed { get; init; }
+    public long SkipDayCost { get; init; }
+    public IList<AttendanceDayItem> Days { get; set; }
+
+    public AttendGift() {
+        Days = new List<AttendanceDayItem>();
+    }
+
+    public override void WriteTo(IByteWriter writer) {
+        writer.WriteInt(Id);
+        writer.WriteLong(BeginTime);
+        writer.WriteLong(EndTime);
+        writer.WriteUnicodeString(AttendanceName);
+        writer.WriteString(Url);
+        writer.WriteByte();
+        writer.WriteBool(DisableClaimButton);
+        writer.WriteInt(TimeRequired);
+        writer.WriteByte();
+        writer.WriteInt();
+        writer.Write<AttendGiftCurrencyType>(SkipDayCurrencyType);
+
+        if (SkipDayCurrencyType != AttendGiftCurrencyType.None) {
+            writer.WriteInt(SkipDaysAllowed);
+            writer.WriteLong(SkipDayCost);
+            writer.WriteInt();
+        }
+
+        writer.WriteInt(Days.Count);
+        foreach (AttendanceDayItem day in Days.OrderBy(day => day.Day)) {
+            writer.WriteClass<AttendanceDayItem>(day);
+        }
+    }
+    
+    public class AttendanceDayItem : IByteSerializable{
```

Comment:
Was this intentionally a nested class? If so, maybe you can use a name that's less redundant since it will always be scoped with `AttendGift.`